api/gateway/stripe/stripe.cfc

```diff
@@ -434,16 +593,56 @@
 		<cfreturn post />
 	</cffunction>
 
+
+	<cffunction name="addBankAccount" output="false" access="private" returntype="any" hint="Add payment source fields to the request object">
+		<cfargument name="post" type="struct" required="true" />
+		<cfargument name="account" type="any" required="true" />
+
+		<cfscript>
+			post["bank_account[country]"] = arguments.account.getCountry();
+			post["bank_account[routing_number]"] = arguments.account.getRoutingNumber();
+			post["bank_account[account_number]"] = arguments.account.getAccount();
+		</cfscript>
+	
+		<cfreturn post />
+	</cffunction>
+	
+
 	<cffunction name="addToken" output="false" access="private" returntype="any" hint="Add payment source fields to the request object">
 		<cfargument name="post" type="struct" required="true" />
 		<cfargument name="account" type="any" required="true" />
 		
 		<!--- required when using as a payment source --->
-		<cfset arguments.post["customer"] = arguments.account.getID() />
+		<cfif findNoCase("cus_", arguments.account.getID())>
+			<cfset arguments.post["customer"] = arguments.account.getID() />
+		<cfelse>
+			<cfset arguments.post["source"] = arguments.account.getID() />
+		</cfif>
 
 		<cfreturn arguments.post />
 	</cffunction>
 
+
+	<cffunction name="addLegalEntity" output="false" access="private" returntype="any" hint="Add payment source fields to the request object">
+		<cfargument name="post" type="struct" required="true" />
+		<cfargument name="account" type="any" required="true" />
+
+		<cfscript>
+			post["card[number]"] = arguments.account.getAccount();
```

Comment:
In my reading of the docs, you don't provide card* fields when creating an account:  https://stripe.com/docs/api/curl#create_account  The legal_entity key looks to be like:

```
"legal_entity": {
    "type": "corporation",
    "business_name": "MegaCorp, Inc.",
    "address": {
      "line1": "123 Anywhere Lane",
      "line2": null,
      "city": "Vancouver",
      "state": "BC",
      "postal_code": "V6Z 2D3",
      "country": "CA"
    },
    "first_name": "Frank",
    "last_name": "Smith",
    "personal_address": {
      "line1": "123 Anywhere Lane",
      "line2": null,
      "city": "Vancouver",
      "state": "BC",
      "postal_code": "V6Z 2D3",
      "country": "CA"
    },
    "dob": {
      "day": 1,
      "month": 1,
      "year": 1980
    },
    "additional_owners": null,
    "verification": {
      "status": "unverified",
      "document": null,
      "details": null
    }
  }
```